onboarding/source/amb_light_service.c

```diff
@@ -1,39 +1,134 @@
 #include "amb_light_service.h"
 #include "serial_io.h"
+#include <stdio.h>
 
 #include <adc.h>
 #include <sci.h>
+#include <FreeRTOS.h>
+#include <os_task.h>
+
+#include <os_queue.h>
+#include <os_projdefs.h>
+#include <string.h>
+
+#define LIGHT_SENSOR 	6U
 
 /* USER CODE BEGIN */
 // Include any additional headers and global variables here
 
+static TaskHandle_t lightServiceHandle = NULL;
+static QueueHandle_t event_queue = NULL;
 /* USER CODE END */
 
 /**
  * @brief Task that waits for MEASURE_LIGHT event and then prints the ambient light value to the serial port.
  * @param pvParameters Task parameters
  */
+
+void adcStartConversion_selChn(adcBASE_t *adc, uint8_t channel, uint8_t fifo_size, uint8_t group);
+void adcGetSingleData(adcBASE_t *adc, uint8_t group, adcData_t *data);
+uint32_t getLightSensorData(void);
 static void lightServiceTask(void * pvParameters);
 
 uint8_t initLightService(void) {
     /* USER CODE BEGIN */
     // Create the task and queue here.
+    BaseType_t xReturned = pdFAIL;
+    
+    if (event_queue == NULL) {
+        event_queue = xQueueCreate(1, LIGHT_SERVICE_QUEUE_SIZE);
```

Comment:
The first argument should be LIGHT_SERVICE_QUEUE_SIZE (i.e. the length of the queue).